src/twisted/conch/ssh/session.py

```diff
@@ -91,6 +92,33 @@ def request_pty_req(self, data):
         else:
             return 1
 
+    def request_env(self, data):
+        """
+        Process a request to pass an environment variable.
+
+        @param data: The environment variable name and value, each encoded
+            as an SSH protocol string and concatenated.
+        @type data: L{bytes}
+        @return: A true value if the request to pass this environment
+            variable was accepted, otherwise a false value.
+        """
+        if not self.sessionSetEnv:
+            try:
+                self.sessionSetEnv = ISessionSetEnv(self.avatar)
+            except TypeError:
+                log.msg('unhandled request for env')
```

Comment:
It makes me a bit sad to see a new old-style log message get added like this 😞 .  But I guess we don't have a great "list of migrations in progress" document that reviewers should look at, so it's not surprising it got missed.

To make a new-style log message, make a new `twisted.logger.Logger` instance at module scope, and call `log.<appropriate level method>`.  In this case, probably `log.info`.

Also: unhandled request for *what* env?  Of whom?  All log messages should include relevant information that will make them actionable by administrators, or at least by the implementor of the avatar class.  In this case I'd say something like `_log.info('SSH avatar {avatar} does not handle setting requested environment variables {env}', avatar=self.avatar, env=data)` or similar.

Reply:
Is there a particular reason to place the log at the module scope? It would be more helpful to me for the logger to be at a class attribute so the exact instance can be known from `log_source` in the event. In my case I want to associate logs from a certain SSH connection to the "owner" of that connection.

Reply:
@ryban I think we can address that as part of [#9773](https://twistedmatrix.com/trac/ticket/9773)